src/common/browser-adapters/webextension-browser-adapter.ts

```diff
@@ -14,29 +17,37 @@ import browser, {
 import { BrowserAdapter } from './browser-adapter';
 import { CommandsAdapter } from './commands-adapter';
 import { StorageAdapter } from './storage-adapter';
-
 export abstract class WebExtensionBrowserAdapter
     implements BrowserAdapter, StorageAdapter, CommandsAdapter
 {
     constructor(private readonly browserEventManager: BrowserEventManager) {}
 
-    public initialize(browserEvents: DictionaryStringTo<Events.Event<any>>): void {
-        // Add browser listeners synchronously on initialization
-        for (const [eventType, event] of Object.entries(browserEvents)) {
-            this.browserEventManager.addBrowserListener(event, eventType);
-        }
-    }
-
     public abstract getManageExtensionUrl(): string;
 
+    public allRequiredEvents(): DictionaryStringTo<Events.Event<any>> {
+        return {
+            TabsOnActivated: browser.tabs?.onActivated,
```

Comment:
We originally restructured things in part to avoid having all of these undefined checks that could cause odd errors, right? I would prefer to keep the browser-event-provider or something analogous so we don't have to have all of these `browser.foo?...` checks here.

Reply:
The `?` checks weren't actually necessary here, they were a relic from an earlier attempt that didn't use a containing function. Removed them!